src/cursor/cur_backup.c

```diff
@@ -180,6 +229,101 @@ __wt_curbackup_open(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *other,
     return (ret);
 }
 
+/*
+ * __backup_get_ckpt --
+ *     Get the most recent checkpoint information and store it in the structure.
+ *
+ * XXX - Currently set return to static void for the compiler, when this function has real content
+ *     it should be static int.
+ */
+static void
+__backup_get_ckpt(WT_SESSION_IMPL *session, WT_BLKINCR *incr)
+{
+    WT_UNUSED(session);
+    WT_UNUSED(incr);
+    /*
+     * Look up the most recent checkpoint and store information about it in incr.
+     *
+     * XXX When this function has content, return a real value. return (0);
+     */
+    return;
+}
+
+/*
+ * __backup_add_id --
+ *     Add the identifier for block based incremental backup.
+ */
+static int
+__backup_add_id(WT_SESSION_IMPL *session, WT_CONFIG_ITEM *cval)
+{
+    WT_BLKINCR *blk;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    u_int i;
+
+    conn = S2C(session);
+    blk = NULL;
+    for (i = 0; i < WT_BLKINCR_MAX; ++i) {
+        blk = &conn->incr_backups[i];
+        /* If it isn't use we can use it. */
+        if (!F_ISSET(blk, WT_BLKINCR_INUSE)) {
+            if (blk->id != NULL)
+                __wt_verbose(session, WT_VERB_BACKUP,
+                  "Freeing and reusing backup slot with old id %s", blk->id);
+            /* Free any string that was there. */
+            __wt_free(session, blk->id);
+            WT_ERR(__wt_strndup(session, cval->str, cval->len, &blk->id));
+            __wt_verbose(session, WT_VERB_BACKUP, "Using backup slot %u for id %s", i, blk->id);
+            /*
+             * XXX
+             *
+             * WT_ERR(__backup_get_ckpt(session, blk));
+             */
+            __backup_get_ckpt(session, blk);
+            F_SET(blk, WT_BLKINCR_VALID);
+            return (0);
+        }
+    }
+    /*
+     * We didn't find an entry. This should not happen.
+     */
+    ret = WT_NOTFOUND;
+err:
+    if (blk != NULL)
+        __wt_free(session, blk->id);
+    return (ret);
```

Comment:
If you fall out of this loop, `blk` won't be NULL so bad stuff will happen.

I'd suggest a stronger message if we can't find a slot for the ID, and "not-found" doesn't seem like the right error.

This might be simpler as:
```
for (i = 0; i < WT_BLKINCR_MAX; ++i)
    if (!F_ISSET(blk, WT_BLKINCR_INUSE))
        break;

if (i == WT_BLKINCR_MAX)
     /* we're in trouble */

/* use the slot */
```

Reply:
Good catch. I will fix.